EpiAware/src/EpiInfModels/Renewal.jl

```diff
@@ -111,6 +116,166 @@ function (epi_model::Renewal)(recent_incidence, Rt)
+prior and a transformation function.
+
+```julia
+using Distributions, Turing, EpiAware
+gen_int = [0.2, 0.3, 0.5]
+g = exp
+
+# Create an EpiData object
+data = EpiData(gen_int, g)
+
+# Create an Renewal model
+renewal_model = RenewalWithPopulation(data = data, initialisation_prior = Normal(), pop_size = 1e6)
+```
+
+Then, we can use `generate_latent_infs` to construct a Turing model for the unobserved
+infection generation model set by the type of `direct_inf_model`.
+
+```julia
+# Construct a Turing model
+Z_t = randn(100) * 0.05
+latent_inf = generate_latent_infs(renewal_model, Z_t)
+```
+
+Now we can use the `Turing` PPL API to sample underlying parameters and generate the
+unobserved infections.
+
+```julia
+# Sample from the unobserved infections model
+
+#Sample random parameters from prior
+θ = rand(latent_inf)
+#Get unobserved infections as a generated quantities from the model
+I_t = generated_quantities(latent_inf, θ)
+```
+"
+@kwdef struct RenewalWithPopulation{S <: Sampleable} <: AbstractRenewal
+    data::EpiData
+    initialisation_prior::S = Normal()
+    pop_size::Float64 = 1e6
+end
+
+@doc """
+    function (epi_model::RenewalWithPopulation)(recent_incidence_and_available_sus, Rt)
+
+Callable on a `RenewalWithPopulation` struct for compute new incidence based on
+recent incidence, Rt and depletion of susceptibles.
+
+## Mathematical specification
+
+The new incidence is given by
+
+```math
+I_t = {S_{t-1} / N} R_t \\sum_{i=1}^{n-1} I_{t-i} g_i
+```
+
+where `I_t` is the new incidence, `R_t` is the reproduction number, `I_{t-i}` is the recent incidence
+and `g_i` is the generation interval.
+
+# Arguments
+- `recent_incidence_and_available_sus`: A tuple with an array of recent incidence
+values and the remaining susceptible/available individuals.
+- `Rt`: Reproduction number.
+
+# Returns
+- Tuple containing the updated incidence array and the new `recent_incidence_and_available_sus`
+value.
+"""
+function (epi_model::RenewalWithPopulation)(recent_incidence_and_available_sus, Rt)
+    recent_incidence, S = recent_incidence_and_available_sus
+    new_incidence = max(S / epi_model.pop_size, 0.0) * Rt *
+                    dot(recent_incidence, epi_model.data.gen_int)
+    new_S = S - new_incidence
+    new_recent_incidence_and_available_sus = (
+        [new_incidence; recent_incidence[1:(epi_model.data.len_gen_int - 1)]], new_S)
+
+    return (new_recent_incidence_and_available_sus, new_incidence)
+end
+
+"""
+Constructs the initial conditions for a renewal model with population.
+
+# Arguments
+- `epi_model::RenewalWithPopulation`: The renewal model with population.
+- `I₀`: The initial number of infected individuals.
+- `Rt`: The time-varying reproduction number.
+
+# Returns
+- A tuple containing the initial number of infected individuals at each generation
+interval and the population size of susceptible/available people.
+
+"""
+function make_renewal_init(epi_model::RenewalWithPopulation, I₀, Rt)
```

Comment:
This duplication is what worries me. Shouldn't this just be based on AbstractRenewal?

Reply:
No, the different models are based on different states so get different `init`s

This model has a state $((I_{t-1}, I_{t-2}, ..., I_{t-m}), S_{t-1})$

Reply:
I see... can we not nest this better?

Reply:
Obviously, we can get rid of this by looking at whole infection time series but I don't like that (very inefficient) but I can be persuaded

Reply:
yeah don't love that either. Looking at this its just that one has I_0 at the end right? So why can't the internals be otherwise shared?